cmd/kueue/main.go

```diff
@@ -292,7 +293,12 @@ func setupProbeEndpoints(mgr ctrl.Manager) {
 	// the function, otherwise a not fully-initialized webhook server (without
 	// ready certs) fails the start of the manager.
 	if err := mgr.AddReadyzCheck("readyz", func(req *http.Request) error {
-		return mgr.GetWebhookServer().StartedChecker()(req)
+		select {
+		case <-certsReady:
```

Comment:
When we discussed this originally I wasn't sure if this is needed https://github.com/kubernetes-sigs/kueue/pull/1676#discussion_r1474188770. And I'm still not 100% sure, but the change reflects the intention, and it should be harmless. Also, the analogous change is done in another project: https://github.com/kubernetes-sigs/hierarchical-namespaces/blob/d367fc08a261135b63d22aeb01c688317d9b7e02/cmd/manager/main.go#L296-L307, so I'm happy to try this fix.

Reply:
@trasc can you give a rationale on why you think this fixes the issue?

From the logs, it does look like there is some condition between setting the certs and starting the manager, so I can see the relationship. But the specific set of events are unclear to me.

Reply:
From the old thread: https://github.com/kubernetes-sigs/kueue/pull/1676#discussion_r1473991998:
```
What I verified experimentally is that if I just call GetWebhookServer before WaitForCertsReady then the webhook is not initialized, and it stays that way.
```
I did this experiments just by modifying the code and calling GetWebhookServer before or after https://github.com/kubernetes-sigs/kueue/blob/c4a2255831a9284a0e6cc213d5938556ea209f49/cmd/kueue/main.go#L226. 

I think what might be now happening when the readiness probe is called between the readiness server is listening (probably does not require certs) and the certs are initialized.

Reply:
> @trasc can you give a rationale on why you think this fixes the issue?
> 
> From the logs, it does look like there is some race condition between setting the certs and starting the manager, so I can see the relationship. But the specific set of events are unclear to me.

Unfortunately I don't have anything concert, I suspect that it the original case, the Start hit in the middle of a rotation. But it's just a suspicion.